### PR TITLE
VACMS-17885: Create facility services links on VBA Facility page

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #17885 

## Testing done
Manually

## Screenshots
Screenshot 2024-07-02 at 2 56 56 PM

### Set up QA Content Publisher
- [x] As an admin, update the QA Content Editor
  - [x] Assign the following roles
    - [x] Content creator - VBA
    - [x] Content editor
  - [x] Assign the following Section
    - [x] Portland VA Regional Benefit Office
    - [x] Cheyenne VA Regional Benefit Office

## Create a service in Portland VA Regional Benefit Office
- [x] As QA Content Editor, go to the [Portland VA Regional Benefit Office](https://pr18442-8wvl5xydl5ab9vkai4vfloiff5nvuuoy.ci.cms.va.gov/node/4237/edit)
- [x] Click on **Create a new service for this facility (opens in new window).**
- [x] Confirm that the Section is set to "Portland VA Regional Benefit Office"
- [x] Confirm that the Facility is "Portland VA Regional Benefit Office"
- [x] Create a new service (such as Disability compensation)
- [x] Save the service
- [x] Close the window

## Review the Portland VA Regional Benefit Office facility services
- [x] On the Portland VA Regional Benefit Office page, click on **Manage existing services for this facility (opens in new window)**
- [x] Confirm that the VBA Facilities are listed, filtered for the Portland VA Regional Benefit Office

## Create a service in Cheyenne VA Regional Benefit Office 
- [x] As QA Content Editor, go to the [VA Regional Benefit Satellite Office at Sheridan VA Medical Center](https://pr18442-8wvl5xydl5ab9vkai4vfloiff5nvuuoy.ci.cms.va.gov/node/48920/edit)
- [x] Click on **Create a new service for this facility (opens in new window).**
- [x] Confirm that the Section is set to "Cheyenne VA Regional Benefit Office" 
- [x] Confirm that the Facility is "VA Regional Benefit Satellite Office at Sheridan VA Medical Center"
- [x] Create a new service (such as Disability compensation)
- [x] Save the service
- [x] Close the window

 ## Review the Cheyenne VA Regional Benefit Office facility services
- [x] On the VA Regional Benefit Satellite Office at Sheridan VA Medical Center page, click on **Manage existing services for this facility (opens in new window)**
- [x] Confirm that the VBA Facilities are listed, filtered for the VA Regional Benefit Satellite Office at Sheridan VA Medical Center

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: docroot/modules/custom/va_gov_vba_facility/src/EventSubscriber/VbaFacilitySubscriber.php [hunk 1]
```diff
@@ -136,6 +136,39 @@ public function alterVbaFacilityNodeForm(FormIdAlterEvent $event): void {
...
+    $form_object = $form_state->getFormObject();
+    $entity = $form_object->getEntity();
+    $section_tid = $entity->field_administration->target_id;
+    $facility_nid = $entity->nid->value;
+    $add_new_service_url = "/node/add/vba_facility_service?field_administration=$section_tid&field_office=$facility_nid";
+    $encoded_facility_name = urlencode($entity->title->value);
+
+    if (isset($form["#fieldgroups"]["group_facility_services"]->format_settings["description"])) {
+      $form["#fieldgroups"]["group_facility_services"]->format_settings["description"] = "
+      <p><a href='$add_new_service_url' target='_blank'>Create a new service for this facility (opens in new window)</a></p>
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_vba_facility/src/EventSubscriber/VbaFacilitySubscriber.php:168]: SHOULD...
...use a more structured approach to the markup here to support readability and ultimately translations. eg:

```php
  $form["#fieldgroups"]["group_facility_services"]->format_settings["description"]['create'] = [
    '#title' => $this->t('Create a new service for this facility (opens in new window)'),
    '#type' => 'link',
    '#url' => // the url,
    '#prefix' => '<p>',
    '#suffix' => '</p>',
  ];
  $form["#fieldgroups"]["group_facility_services"]->format_settings["description"]['manage'] = [
    '#title' => $this->t('Manage existing services for this facility (opens in new window)'),
    '#type' => 'link',
    '#url' => // the url,
    '#prefix' => '<p>',
    '#suffix' => '</p>',
  ];
```

Comment (omahane) [reply docroot/modules/custom/va_gov_vba_facility/src/EventSubscriber/VbaFacilitySubscriber.php:168]: I tried to implement this, but kept getting errors that appear to be related to how ['description'] is expecting a string, not an array.
Screenshot 2024-07-03 at 8 46 11 AM

I wrapped the link text in a t() and made it all a little more readable.

```
$create_service_url = "/node/add/vba_facility_service?field_administration=$section_tid&field_office=$facility_nid";
    $create_service_text = $this->t('Create a new service for this facility (opens in new window)');
    $encoded_facility_name = urlencode($entity->title->value);
    $manage_services_url = "/admin/content?title=$encoded_facility_name&type=vba_facility_service&moderation_state=All&owner=All";
    $manage_services_text = $this->t('Manage existing services for this facility (opens in new window)');

    if (isset($form["#fieldgroups"]["group_facility_services"]->format_settings["description"])) {
      $form["#fieldgroups"]["group_facility_services"]->format_settings["description"] = "
        <p><a href='$create_service_url' target='_blank'>$create_service_text</p>
        <p><a href='$manage_services_url' target='_blank'>$manage_services_text</p>
        ";
    }
```

**Review by dsasser — APPROVED** [review]
Looks good. Both create and manage links are working per ACs.

Screenshot 2024-07-03 at 8 29 09 AM
Screenshot 2024-07-03 at 8 29 02 AM
Screenshot 2024-07-03 at 8 28 59 AM

**Comment by omahane** [comment]
> Works as expected on Node:Edit. Would adding it to Node:View be a significant additional lift?

Yes, it would.

**Pull request merged**